presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -339,6 +340,9 @@ public void testNot()
         assertOptimizedEquals("not 1!=1", "true");
         assertOptimizedEquals("not unbound_long=1", "not unbound_long=1");
         assertOptimizedEquals("not unbound_long=(1+1)", "not unbound_long=2");
+        assertOptimizedEquals("cast(json_parse(unbound_string) as map(varchar, varchar))", "cast(json_parse(unbound_string) as map(varchar, varchar))");
```

Comment:
why these are in `testNot`? -- Also it doesn't seem to perform the optimization?

Reply:
They should be in `testFunctionCall`. There is no good way to test it actually. Just to invoke the codepath to make sure there is no exception